File: chat-demo-android/src/main/java/com/twilio/chat/demo/ToastStatusListener.java
package com.twilio.chat.demo;

import com.twilio.chat.ErrorInfo;
import com.twilio.chat.StatusListener;

class ToastStatusListener extends StatusListener
{
    private final String okText;
    private final String errorText;

    ToastStatusListener(String ok, String error) {
        okText = ok;
        errorText = error;
    }

    @Override
    public void onSuccess()
    {
        TwilioApplication.get().showToast(okText);
    }

    @Override
    public void onError(ErrorInfo errorInfo)
    {
        TwilioApplication.get().showError(errorText, errorInfo);
    }
}

File: chat-demo-android/src/main/java/com/twilio/chat/demo/ToastStatusListener.kt
package com.twilio.chat.demo

import com.twilio.chat.ErrorInfo
import com.twilio.chat.StatusListener

/**
 * Status listener that shows a toast with operation results.
 */
internal open class ToastStatusListener(private val okText: String, private val errorText: String) : StatusListener() {

    override fun onSuccess() {
        TwilioApplication.instance.showToast(okText)
    }

    override fun onError(errorInfo: ErrorInfo?) {
        TwilioApplication.instance.showError(errorText, errorInfo!!)
    }
}
